src/operators/catch.ts

```diff
@@ -50,7 +50,17 @@ class CatchSubscriber<T> extends Subscriber<T> {
     if (result === errorObject) {
       this.destination.error(errorObject.e);
     } else {
-      this.add(result.subscribe(this.destination));
+      this.lastSubscription.unsubscribe();
+      this.lastSubscription = result.subscribe(this.destination);
     }
   }
+
+  _complete() {
+    this.lastSubscription.unsubscribe();
+    this.destination.complete();
+  }
+
+  _unsubscribe() {
```

Comment:
According to test coverage, yes it is. `_unsubscribe()` is overriding Subscription._unsubscribe(), which is called inside Subscription.unsubscribe().